src/utils/context.ts

```diff
@@ -21,7 +23,8 @@ export const DefaultCowContext = {
   appDataHash: DEFAULT_APP_DATA_HASH,
   isDevEnvironment: false,
   ipfs: {
-    uri: DEFAULT_IPFS_GATEWAY_URI,
+    readUri: DEFAULT_IPFS_READ_URI,
+    writeUri: DEFAULT_IPFS_WRITE_URI,
```

Comment:
Is this splitted in two because the read one can use any GATEWAY? (like cloudflare) and the write is used only for pinning and uploading docs

If this is the case, I wonder if we can be more explicit, and also isolate the "oppnionated services" we use, so if we want in the future we can provide more alternatives.

With this I'm referring to make the interface:

```ts
export interface Ipfs {
  gatewayUrl?: string
  ipfsService?: PinataIpfsService | ... leave it open for more services in the future
}

interface PinataIpfsService {
  baseUrl: string
  apiKey: string
  apiSecret?: string
  }
```

Reply:
Will add that as a follow up task

Reply:
https://github.com/cowprotocol/cow-sdk/issues/33